UnityWeld_Editor/AnimatorParameterBindingEditor.cs

```diff
@@ -0,0 +1,337 @@
+            );
+
+            EditorGUILayout.Space();
+
+            EditorStyles.label.fontStyle = viewModelPropertyPrefabModified ? FontStyle.Bold : defaultLabelStyle;
+
+            var adaptedViewPropertyType = AdaptTypeBackward(viewPropertyType, targetScript.ViewAdapterTypeName);
+            ShowViewModelPropertyMenu(
+                new GUIContent("View-model property", "Property on the view-model to bind to."),
+                TypeResolver.FindBindableProperties(targetScript),
+                updatedValue => targetScript.ViewModelPropertyName = updatedValue,
+                targetScript.ViewModelPropertyName,
+                property => property.PropertyType == adaptedViewPropertyType
+            );
+
+            GUI.enabled = guiPreviouslyEnabled;
+
+            EditorStyles.label.fontStyle = defaultLabelStyle;
+
+            EditorGUILayout.Space();
+
+        }
+
+        private void ShowAnimatorParametersMenu(
+            GUIContent label,
+            Action<AnimatorParameterTypeAndName> propertyValueSetter,
+            AnimatorParameterTypeAndName curPropertyValue,
+            out Type selectedPropertyType
+        )
+        {
+            var animator = targetScript.GetComponent<Animator>();
+
+            AnimatorControllerParameter[] properties = new AnimatorControllerParameter[] { };
+
+            // For whatever reason, accessing Animator.parameters while the GameObject the
+            // Animator is on is inactive causes it to return an empty array, aswell as fill
+            // the unity console with warnings. This behaviour is undocumented, and this is just
+            // a work around which seems to work fine
+            if (!animator.gameObject.activeInHierarchy)
+            {
+                UnityEditorInternal.ComponentUtility.CopyComponent(animator.transform);
+                var parent = animator.transform.parent;
+                var siblingIndex = animator.transform.GetSiblingIndex();
+                var isActive = animator.gameObject.activeSelf;
+                animator.transform.SetParent(null);
+                animator.gameObject.SetActive(true);
+                properties = animator.parameters;
+                animator.transform.SetParent(parent);
+                animator.transform.SetSiblingIndex(siblingIndex);
+                animator.gameObject.SetActive(isActive);
+                UnityEditorInternal.ComponentUtility.PasteComponentValues(animator.transform);
+            }
+            else
+            {
+                properties = animator.parameters;
+                //Another odd fix to refresh the parameters
+                if (properties == null || properties.Length == 0)
+                {
+                    var isActive = animator.gameObject.activeSelf;
+                    animator.gameObject.SetActive(false);
+                    animator.gameObject.SetActive(true);
+                    animator.gameObject.SetActive(isActive);
```

Comment:
Toggling the object active does seem like an odd fix. I think it would be worth expanding the comment a bit to explain why it is necessary/what breaks if we don't do it this way.

Reply:
Yeah, not a great comment sorry. I forgot exactly what this fixes, which just shows the comment needs to be better lol. I'll look into why I am doing this